mirror/cache.go

```diff
@@ -38,25 +39,32 @@ func (c *Cache) isExpired() bool {
 
 func (c *Cache) refresh() error {
 	var fs []File
-	s, err := session.NewSession(&aws.Config{
-		Region:           aws.String(c.settings.region),
-		Endpoint:         aws.String(c.settings.endpointURL),
-		S3ForcePathStyle: aws.Bool(true),
-		Credentials: credentials.NewStaticCredentials(
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(c.settings.region),
+		config.WithEndpointResolverWithOptions(aws.EndpointResolverWithOptionsFunc(
+			func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+				return aws.Endpoint{
+					URL:               c.settings.endpointURL,
+					HostnameImmutable: true,
+				}, nil
+			})),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
 			c.settings.accessKey,
 			c.settings.secretAccessKey,
 			"",
-		),
-	})
+		)),
+	)
 	if err != nil {
 		return err
 	}
 
 	var t *string
 	loadPage := func(t *string) ([]File, *string, error) {
 		var fs []File
-		sdk := s3.New(s)
-		r, err := sdk.ListObjectsV2(&s3.ListObjectsV2Input{
+		client := s3.NewFromConfig(cfg, func(o *s3.Options) {
```

Comment:
Why using a ling variable name? Can you think of something shorter as we had before?